lib/buffer.js

```diff
@@ -567,9 +567,8 @@ Buffer.prototype.copy = function(target, targetStart, sourceStart, sourceEnd) {
 // This behaves neither like String nor Uint8Array in that we set start/end
 // to their upper/lower bounds if the value passed is out of range.
 Buffer.prototype.toString = function(encoding, start, end) {
-  var result;
   if (arguments.length === 0) {
-    result = this.utf8Slice(0, this.length);
+    return this.utf8Slice(0, this.length);
```

Comment:
By changing this to `return`, you can drop the `else` and save a level of indentation on the rest of the code below.